zqd/storage/archivestore/archivestore.go

```diff
@@ -113,10 +113,26 @@ func (s *Storage) Summary(_ context.Context) (storage.Summary, error) {
 	return sum, nil
 }
 
+func (s *Storage) Write(ctx context.Context, zctx *resolver.Context, zr zbuf.Reader) error {
+	_, err := s.ark.UpdateCheck()
```

Comment:
Instead of an UpdateCheck here, I suspect adjustments for locking need to be made to archive.Import and/or archive.Archive.AppendSpans.